Bowling.java

```diff
@@ -113,17 +124,17 @@ public Scoreboard() {
 
 	public void update(Integer pinsDown) {
 		currentFrame().update(pinsDown);
+		updateTotalPoints();
 
 		if (currentFrame().isStrike() && !isFinalFrame()) {
 			currentFrame().update(0);
 		}
 
 		if (currentFrame().isFull()) {
-			updateTotalPoints();
 			
 			if (allFrames.size() == 9) {
-				addFrame(true);
-			} else {
+				addFinalFrame();
+			} else if (allFrames.size() < 9) {
```

Comment:
I'm always concerned when I see if checks that don't cover all cases.  It makes perfect sense in this case, if we're ever at 10 frames we shouldn't be adding another frame.  However, my knee jerk reaction was to suggest that this if check isn't needed.

Oh the fun with self-analysis.